iOS-Email-Client/Managers/DBManager.swift

```diff
@@ -327,10 +327,11 @@ class DBManager: SharedDB {
             }
             thread.unread = threadEmails.contains(where: {$0.unread})
             thread.counter = threadEmails.count
-            thread.subject = threadEmails.first!.subject
+            thread.subject = threadEmails.last!.subject
             threadIds.insert(thread.threadId)
             threads.append(thread)
         }
+        threads.sort { $0.date > $1.date }
```

Comment:
Sorting after quering the data might end up into some inconsistent element orders and pagination problems, we should do this before applying custom distinct